modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.hpp

```diff
@@ -79,6 +78,23 @@ class Jpeg2KOpjDecoder CV_FINAL : public BaseImageDecoder
     detail::OpjMemoryBuffer opjBuf_;
 
     OPJ_UINT32 m_maxPrec = 0;
+    OPJ_CODEC_FORMAT format_;
+};
+
+class Jpeg2KJP2OpjDecoder CV_FINAL : public Jpeg2KOpjDecoderBase {
+public:
+    Jpeg2KJP2OpjDecoder();
+     ~Jpeg2KJP2OpjDecoder() CV_OVERRIDE = default;
+
+    ImageDecoder newDecoder() const CV_OVERRIDE;
+};
+
+class Jpeg2KJ2KOpjDecoder CV_FINAL : public Jpeg2KOpjDecoderBase {
```

Comment:
> J2K

j2c ?

Reply:
The OpenJPEG codec constant for codestreams is called `OPJ_CODEC_J2K` so I'm using `J2K` here for consistency.
Codestreams don't have an official extension and have several common extensions, like .j2k, .j2c and a few others.